### ISSUE TITLE
Default behaviour for autotune when val_path = None

### ISSUE THREAD

**Issue by pedrojuanbj** [CSML-IIT-UCL/franken#42]
Hello, 

Just a suggestion due to some heavy testing I have been doing so far. 

I believe the default behaviour from autotune now when one wants to train and does not provide a validation dataset path is to use the model that performs best on the training data. However I do believe this is a mistake and it really can make the models overfitted. A much better behaviour would be to approach this as it is done in MACE where if you don't provide a val_path it splits the training data in two with a part dedicated to validation. 

I understand that one can simply just split the data themselves, but sometimes, especially if someone is not so familiar with the code may expect this behaviour already coming from other softwares or think models are really bad being misled by the overfitting. 

For example in this case, Up to 80 structures I used 20 as validation, where at 100 training structures I used NO validation, and it really worsened even though there was more data to train from. For some models this is more noticeable and for other systems the same model underperfrorming can in turn be better with more data and no validation. Best practice would be to always save for validation, otherwise it would lead to heavy overfitting in most cases.  

<img width="1331" height="557" alt="Image" src="https://github.com/user-attachments/assets/fd0918b2-f540-4854-bdc7-d23b5c1c0a6d" />

**Comment by luigibonati** [comment]
Hi Pedro, thank you very much for your detailed feedback.

I think there may be some confusion in the current behavior, and we could definitely improve this, either through better documentation or at least some warning in the CLI

From my understanding, a validation set is primarily needed for hyperparameter optimization and model selection. Once the best hyperparameters are identified, one could in principle retrain using the full dataset. However, this assumes that the hyperparameters are fixed and not being changed without a validation signal, otherwise as you pointed out it will overfit. 

Maybe rather than automatically splitting the dataset we should issue a warning when no validation set is provided but an hyperparameter optimization search is performed?

PS:  how are you evaluating the model performance in your comparison in the image? Is this based on an independent test set?

**Comment by pedrojuanbj** [comment]
Hi Luigi, 

Thanks a lot for your prompt reply. 

I do believe issuing a big warning in both CLI and API would be in place regardless. One of the strengths of the code we shipped is in the ease of use and the autotune module, which is what one should use when approaching a new system and how most people would be introduced to the method (even the begginer tutorial). We've tried to make it more approachable to end users / application people and they might not be so familiar with some of these concepts like train/val/test, most "begginer ML" packages like sklearn, keras  etc. only have train and testing and the validation is done automatically, this is why I mentioned it in the first place, remembering those packages. Also even if its recommended practice to figure out the hyperparameters first and then train the final model, it would require an extra step taking a look at the hyperparameters manually and etc. which could be a bit cumbersome for someone novice. 

Perhaps a compromise would be: 

- if val_path=None --> Warning and allow an option for val_path to be either None or a number between 0 and 1 which would be the percentage split of data for training and testing.

-  If it's too confusing maybe instead add a val_split parameter that does this instead when val_path is None, then if val_split=None and val_path=None just a big warning.  

> PS: how are you evaluating the model performance in your comparison in the image? Is this based on an independent test set?

This was forces RMSE in meV/Ang for an independent test set indeed for 4 models trained on the same labels at different levels.
